plugin.video.ripe/resources/lib/ripe_addon.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+
+'''
+   KODI RIPE Meetings video add-on.
+   Copyright (C) 2020 José Antonio Montes (jamontes)
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+   This is the first trial of the RIPE Meetings video add-on for KODI.
+   This add-on gets the videos from RIPE Meetings web sites and shows them properly ordered.
+   This add-on depends on the lutil library functions.
+'''
+
+from resources.lib.plugin import Plugin
+import resources.lib.ripe_api as api
+
+plugin_id = 'plugin.video.ripe'
+
+localized_strings =  {
+        'Next page': 30010,
+        'Type not suported': 30011,
+        'Video not located': 30012,
+        'Previous page': 30013,
+        'All videos': 30014,
+        'Search': 30015,
+        }
+
+p = Plugin(plugin_id)
+
+settings = p.get_plugin_settings()
+translation = p.get_plugin_translation()
+
+debug_flag = settings.getSetting("debug") == "true"
+
+p.set_debug_mode(debug_flag)
+api.set_debug(debug_flag, p.log)
+
+def get_located_string(string_name):
+    return translation(localized_strings.get(string_name)).encode('utf-8') or string_name if string_name in localized_strings else string_name
+
+
+# Entry point
+def run():
+    p.log("ripe.run")
+
+    # Get params
+    params = p.get_plugin_parms()
+
+    action = params.get("action", '')
+    if action:
+        eval("%s(params)" % action)
```

Comment:
This kinda feels like a security risk, as it will take an input parameter and then executes it within Python. So anyone could basically call your add-on and have any Python code executed. As there are only two options: `main_list` and `play_video`, it would be better to just check for these to options and call the methods.

Reply:
OK, no problem.